modules/nextflow/src/main/groovy/nextflow/forch/ForchTaskHandler.groovy

```diff
@@ -0,0 +1,93 @@
+package nextflow.forch
+
+import java.util.concurrent.TimeUnit
+
+import groovy.json.JsonBuilder
+import groovy.util.logging.Slf4j
+import nextflow.executor.BashWrapperBuilder
+import nextflow.executor.res.AcceleratorResource
+import nextflow.processor.TaskHandler
+import nextflow.processor.TaskRun
+import nextflow.processor.TaskStatus
+import nextflow.script.ProcessConfig
+import nextflow.util.Escape
+import nextflow.util.MemoryUnit
+
+@Slf4j
+class ForchTaskHandler extends TaskHandler {
+
+    ProcessConfig processConfig
+
+    Integer forchTaskId
+
+
+    ForchTaskHandler(TaskRun task) {
+        super(task)
+
+        this.processConfig = task.processor.config
+    }
+
+    private String getCurrentStatus() {
+        if (this.forchTaskId == null) return
+
+        String command = "forch status ${forchTaskId}"
+        StringBuilder stdout = new StringBuilder(), stderr = new StringBuilder();
+        Process proc = command.execute()
+
+        proc.consumeProcessOutput(stdout, stderr)
+        proc.waitFor(5, TimeUnit.SECONDS)
+
+        return stdout.toString().trim()
+    }
+
+    @Override
+    boolean checkIfRunning() {
+        return this.currentStatus == 'running'
+    }
+
+    @Override
+    boolean checkIfCompleted() {
+        return this.currentStatus == 'succeeded' || this.currentStatus == 'failed'
+    }
+
+    @Override
+    void kill() {
+        // noop
```

Comment:
fyi you can insert a `container-killed` event (if it's running) or set `cancelled` on the task (if it's not running) to kill it

see `pods-stop` in nucleus